test-framework/core/src/main/java/org/keycloak/testframework/admin/AdminClientSupplier.java

```diff
@@ -35,16 +33,14 @@ public Keycloak getValue(InstanceContext<Keycloak, InjectAdminClient> instanceCo
 
         InjectAdminClient.Mode mode = annotation.mode();
 
-        KeycloakServer server = instanceContext.getDependency(KeycloakServer.class);
-        KeycloakBuilder clientBuilder = KeycloakBuilder.builder()
-                .serverUrl(server.getBaseUrl())
+        AdminClientBuilder adminBuilder = instanceContext.getDependency(AdminClientFactory.class).create()
                 .grantType(OAuth2Constants.CLIENT_CREDENTIALS);
 
         if (mode.equals(InjectAdminClient.Mode.BOOTSTRAP)) {
-            clientBuilder.realm("master").clientId(Config.getAdminClientId()).clientSecret(Config.getAdminClientSecret());
+            adminBuilder.realm("master").clientId(Config.getAdminClientId()).clientSecret(Config.getAdminClientSecret());
         } else if (mode.equals(InjectAdminClient.Mode.MANAGED_REALM)) {
             ManagedRealm managedRealm = instanceContext.getDependency(ManagedRealm.class);
```

Comment:
When using the managed realm mode, you should be able to specify which realm with `realmRef`.